static/app/utils/discover/genericDiscoverQuery.tsx

```diff
@@ -195,6 +205,32 @@ class _GenericDiscoverQuery<T, P> extends React.Component<Props<T, P>, State<T>>
     );
   };
 
+  /**
+   * The error type isn't consistent across APIs. We see detail as just string some times, other times as an object.
+   */
+  _parseError = (error: any): QueryError | null => {
```

Comment:
can we maybe reuse the function https://github.com/getsentry/sentry/blob/master/static/app/utils/handleXhrErrorResponse.tsx?

Reply:
That's a good catch on that util function, I might give it a shot 👍